src/logic/hooks/useExecutionStatus.ts

```diff
@@ -24,8 +23,7 @@ export const useExecutionStatus = ({
 
   const [status, error, loading] = useAsync(async () => {
     if (!isExecution || !txData) return EstimationStatus.SUCCESS
-    const isEstimationPending =
-      !gasLimit || gasLimit === DEFAULT_GAS_LIMIT || gasPrice === DEFAULT_GAS || gasMaxPrioFee === DEFAULT_GAS
+    const isEstimationPending = !gasLimit || gasPrice === DEFAULT_GAS || gasMaxPrioFee === DEFAULT_GAS
```

Comment:
Not sure I get the fix.
Checking that gas was 0 was wrong, right? Because it can return/error out with 0, and thus not pending.
But you're still checking for 0 here? Also what's the difference between DEFAULT_GAS_LIMIT and DEFAULT_GAS?

Reply:
`DEFAULT_GAS_LIMIT` is `"0"`. Don't we want to make sure that it's not this?

Reply:
`gasLimit === DEFAULT_GAS_LIMIT` was true because if the `useEstimateGasLimit` hits an error it will return `DEFAULT_GAS_LIMIT` which is `'0'`
So the estimation actually finished with an error and is not loading anymore.


It's all a bit confusing because the involved hooks work quite different:
- useEstimateGasLimit returns a string if successfull / error or undefined while still loading
- useEstimateTransactionGas never returns undefined. It returns the DEFAULT_GAS ("0") while loading and on error it returns the MAX_GAS_FEE and MAX_GAS_PRIO. That's why here assuming "0" === LOADING is correct.